runner/src/pipeline.rs

```diff
@@ -37,25 +49,25 @@ impl LayerFunction {
         let mut takes_original_input_kwarg = false;
         let mut takes_context_kwarg = false;
         let parameters = PyIterator::from_object(
-            inspect
+            &inspect
                 .getattr(intern!(py, "signature"))?
-                .call1((func,))?
+                .call1((&func,))?
                 .getattr(intern!(py, "parameters"))?
                 .call_method0(intern!(py, "values"))?,
         )?;
         for parameter in parameters {
             let parameter = parameter?;
             let kind = parameter.getattr(intern!(py, "kind"))?;
-            if kind.eq(positional_only)? || kind.eq(var_positional)? {
+            if kind.eq(positional_only.clone())? || kind.eq(var_positional.clone())? {
```

Comment:
it looks like `&Bound<'py, T>` supports `IntoPyObject<'py>` so you should be able to do `kind.eq(&positional_only)` https://docs.rs/pyo3/latest/pyo3/conversion/trait.IntoPyObject.html#impl-IntoPyObject%3C'py%3E-for-%26Bound%3C'py,+T%3E

Reply:
same goes for the other ones obviously